### ISSUE TITLE
An interaction between pytest fixtures and @example broke in Hypothesis 3.7.3

### ISSUE THREAD

**Issue by alexwlchan** [HypothesisWorks/hypothesis-python#560]
Reported by ronan in IRC. [Part of the PyPy](http://buildbot.pypy.org/summary/longrepr?testname=test_composition%5BNFC-NFC-NFC%5D&builder=own-linux-x86-64&build=5881&mod=pypy.module.unicodedata.test.test_hyp) test suite started failing in Hypothesis 3.7.3.

I’ve reduced the failing behaviour in that test case to the following minimal example, which fails reliably on Python 3.6 for me. It fails in Hypothesis 3.7.3 but not 3.7.2.

```python
import pytest
from hypothesis import example, given
from hypothesis.strategies import integers

@pytest.fixture
def my_fixture():
    pass

@example(xs=1)
@given(xs=integers())
def test_case(xs, my_fixture):
    pass
```

Playing with git bisect points to a5c1200 as the culprit, although that’s such a big patch that I don’t have time to dig into it further right now.

**Comment by DRMacIver** [comment]
Ha ha, what idiot said it seemed pointless to do patch releases for pure refactorings?

(Oh, right, it was me)

Double blame on past-@DRMacIver: I really should have broken that commit up. :-(

**Comment by alexwlchan** [comment]
…we appear not to have any tests around pytest fixtures.

**Comment by alexwlchan** [comment]
Okay, I’ve narrowed it down to the introduction of this chunk in a5c1200:

```python
random = get_random_for_wrapped_test(test, wrapped_test)

test_runner, search_strategy = process_arguments_to_given(
    wrapped_test, arguments, kwargs, generator_arguments,
    generator_kwargs, argspec, test, settings
)

execute_explicit_examples(
    test_runner, test, wrapped_test, settings, arguments, kwargs
)
```

**Comment by DRMacIver** [comment]
> …we appear not to have any tests around interactions with pytest fixtures?

Apparently not. 😢 

We use parametrize + given quite a lot, which implicitly tests similar paths, and we have a bunch of tests that assert the shape of the argspec. But evidently that's not enough.

**Comment by alexwlchan** [comment]
And it’s not this line:

```python
random = get_random_for_wrapped_test(test, wrapped_test)
```

Getting closer…

**Comment by Zac-HD** [comment]
Closed by #562, fix released as Hypothesis 3.7.4
